### ISSUE TITLE
Timeout when refreshing state of storage account

### ISSUE THREAD

**Issue by jwueste** [terraform-providers/terraform-provider-azurerm#4481]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.11.13
provider.azurerm v1.34.0

### Affected Resource(s)

* `azurerm_storage_account`

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "strg" {
  name                      = "${var.name}"
  resource_group_name       = "${azurerm_resource_group.strg.name}"
  location                  = "${azurerm_resource_group.strg.location}"
  account_tier              = "Standard"
  account_kind              = "StorageV2"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true

  network_rules {
    default_action             = "Allow"
    virtual_network_subnet_ids = ["${azurerm_subnet.subnet.id}"]
    bypass                     = ["AzureServices"]
    ip_rules                   = ["${var.allowedIP}"]
  }
}
```

### Expected Behavior

As nothing is changed within the storage resource, the state should only get refreshed and other changes should be done.

### Actual Behavior

Refreshing state fails with a timeout and errormessage.
```
2019-10-02T09:33:22.2239161Z [1m[31mError: [0m[0m[1mError refreshing state: 1 error(s) occurred:
2019-10-02T09:33:22.2239274Z 
2019-10-02T09:33:22.2239324Z * azurerm_storage_account.strg: 1 error(s) occurred:
2019-10-02T09:33:22.2239429Z 
2019-10-02T09:33:22.2240078Z * azurerm_storage_account.strg: azurerm_storage_account.strg: Error reading the advanced threat protection settings of AzureRM Storage Account "accountname": security.AdvancedThreatProtectionClient#Get: Failure responding to request: StatusCode=504 -- Original Error: autorest/azure: Service returned an error. Status=504 Code="GatewayTimeout" Message="The gateway did not receive a response from 'Microsoft.Security' within the specified time period."[0m
2019-10-02T09:33:22.2240421Z 
```
### Steps to Reproduce

1. `terraform apply`

### Important Factoids

The error is not reproducable everytime. After some minutes/hours the same deployment finishes without any error.

**Comment by TheStanHo** [comment]
I am also receiving this error today. When I ran it yesterday it was fine. As a temporary fix I've downgraded my provider to 1.31.0 and it has ran fine.

**Comment by tombuildsstuff** [comment]
hi @jwueste @TheStanHo 

Thanks for opening this issue.

Unfortunately this is an issue within the Azure API where the Internal API isn't responding to the Public Azure API within a designated time period - as such unfortunately there's not a lot we can do to work around this. As was mentioned in #4480 it appears this issue is intermittent - as such I'm going to suggest opening an issue with Microsoft Support (which we'll do on our side too) - but for the moment I'm going to close this issue since this is an issue with the Azure API, rather than something specific to Terraform.

Thanks!
